pkg/worker/faas_http_worker.go

```diff
@@ -108,21 +109,28 @@ func (h *FaasHttpWorker) HandleHttpRequest(trigger *triggers.HttpRequest) (*trig
 	}()
 
 	var mimeType string = ""
-	if trigger.Header != nil {
-		mimeType = trigger.Header["Content-Type"]
+	if trigger.Header != nil && len(trigger.Header["Content-Type"]) > 0 {
+		mimeType = trigger.Header["Content-Type"][0]
 	}
 
 	if mimeType == "" {
 		mimeType = http.DetectContentType(trigger.Body)
 	}
 
+	headers := make(map[string]*pb.HeaderValue)
+	for k, v := range trigger.Header {
+		headers[k] = &pb.HeaderValue{
+			Value: v,
+		}
+	}
+
 	triggerRequest := &pb.TriggerRequest{
 		Data:     trigger.Body,
 		MimeType: mimeType,
 		Context: &pb.TriggerRequest_Http{
 			Http: &pb.HttpTriggerContext{
 				Path:        trigger.Path,
-				Headers:     trigger.Header,
+				Headers:     headers,
 				Method:      trigger.Method,
 				QueryParams: trigger.Query,
 			},
```

Comment:
we might still want to use old headers here as well, but as we're dropping support for PHP at the moment and it's the only runtime that uses this worker it might not be necessary and we might be able to deprecate this code altogether. Need to discuss.

Reply:
I'll create an item in the backlog to address this item so that we can close out this PR and move forward.